streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -230,6 +230,8 @@ public void onPartitionsRevoked(final Collection<TopicPartition> assignment) {
                 // suspend active tasks
                 suspendTasksAndState();
             } catch (final Throwable t) {
+                // TODO: when this happens, we should still make sure the standby / active tasks list gets updated before returning
```

Comment:
The issue is, when one or more tasks get an error when suspending, we are still calling 

```
private void updateSuspendedTasks() {
        suspendedTasks.clear();
        suspendedTasks.putAll(activeTasks);
        suspendedStandbyTasks.putAll(standbyTasks);
    }
```

as the last step of the function call, and the failed tasks will still be added to the `suspendedTasks` and `suspendedStandbyTasks`, so if they get reassigned it may proceed to execute the resume procedure; and if they do not get reassigned the `close()` call will be executed again. Is that OK?